### ISSUE TITLE
Rule proposal: no-empty-line-separating-import

### ISSUE THREAD

**Issue by gajus** [benmosher/eslint-plugin-import#172]
This is stylistic rule. I often see import statements separated one from another using arbitrary rules, e.g.

```js
import yargs from 'yargs';
import getStdin from 'get-stdin';


import {
    lintFiles,
    lintText
} from './../../utilities';

import {
    failHandler,
    getTargetPaths,
    outputReport
} from './../../utilities/bin';
```

This rule would enforce no spacing between import statements, i.e. 

```js
import yargs from 'yargs';
import getStdin from 'get-stdin';
import {
    lintFiles,
    lintText
} from './../../utilities';
import {
    failHandler,
    getTargetPaths,
    outputReport
} from './../../utilities/bin';
```

**Comment by benmosher** [comment]
Just a heads up: if you want to implement + PR these, I'm more than willing to accept them, but I don't have a lot of time ATM and the style-guide rules are lower priority than getting ready for ESLint 2, infrastructure, etc.

**Comment by evilebottnawi** [comment]
@benmosher maybe rename `max-empty-line-separating-import` with option `number`?

**Comment by jfmengels** [comment]
It's a but of a hack, but you can configure the order rule to do the wanted behavior.

**Comment by benmosher** [comment]
I'm cool either way. Seems reasonable to assume that intent to use a rule like this implies desire to use `order`.

**Comment by ljharb** [comment]
That assumption is incorrect - I do not want ordering but do want imports grouped with no newlines.

**Comment by caub** [comment]
I agree, https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/order.md#newlines-between-ignorealwaysalways-an d-inside-groupsnever lets ignore the `newlines-between` option, but not the `group` one

**Comment by Devansu-Yadav** [comment]
@benmosher @ljharb Is this issue still open to take up? I don't see any PR created for this rule, so I was hoping to take a shot at implementing this if it's something useful to have 😅

For starters, I'm thinking of going through the code for [`newlines-between`](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/order.md#newlines-between-ignorealwaysalways-and-inside-groupsnever) option (especially the `never` value that can be passed) of the `order` rule to understand how to avoid empty lines between imports. I'm also looking to play around with the [AST Explorer](https://astexplorer.net/) to get an idea about the implementation.

Though, I'll need some input from your end to get some clarity on the implementation as well as what options we could configure for this rule 😅

**Comment by ljharb** [comment]
@Devansu-Yadav yup! no PR yet. altho let's make sure we're on the same page about implementation first.

**Comment by Devansu-Yadav** [comment]
> @Devansu-Yadav yup! no PR yet. altho let's make sure we're on the same page about implementation first.

Sure, absolutely!

**Comment by Devansu-Yadav** [comment]
> @Devansu-Yadav yup! no PR yet. altho let's make sure we're on the same page about implementation first.

@ljharb Any updates on how I should be proceeding with the implementation?

**Comment by ljharb** [comment]
What schema did you have in mind?

**Comment by Devansu-Yadav** [comment]
> What schema did you have in mind?

As this rule mostly deals with having no newlines between import statements, having a `count` option to count the no of newlines wouldn't make sense here. Do we want an option like [`groups`](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/order.md#groups-array) defined in the `order` rule to enforce no newlines inside specified groups of import statements but ignore newlines between different import groups? This would work exactly like [`always`](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/order.md#newlines-between-ignorealwaysalways-and-inside-groupsnever) value for the `newlines-between` option defined in the `order` rule. The only difference in having the `groups` option here for this rule would be that it would have a default value of `[]`, meaning that we wouldn't have any import groups specified by default.

I'm not sure if it would be useful to only enforce no newlines inside import groups using the `groups` option 😅. What do you think?

I'm also thinking about whether we should have an option like [`considerComments`](https://github.com/Devansu-Yadav/eslint-plugin-import/blob/main/docs/rules/newline-after-import.md#rule-details) to group together comments along with imports without any newline or it would deviate the purpose of this rule. 
I'm open to any suggestions for the schema we should have for this rule.

**Comment by Devansu-Yadav** [comment]
> > What schema did you have in mind?
> 
> As this rule mostly deals with having no newlines between import statements, having a `count` option to count the no of newlines wouldn't make sense here. Do we want an option like [`groups`](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/order.md#groups-array) defined in the `order` rule to enforce no newlines inside specified groups of import statements but ignore newlines between different import groups? This would work exactly like [`always`](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/order.md#newlines-between-ignorealwaysalways-and-inside-groupsnever) value for the `newlines-between` option defined in the `order` rule. The only difference in having the `groups` option here for this rule would be that it would have a default value of `[]`, meaning that we wouldn't have any import groups specified by default.
> 
> I'm not sure if it would be useful to only enforce no newlines inside import groups using the `groups` option 😅. What do you think?
> 
> I'm also thinking about whether we should have an option like [`considerComments`](https://github.com/Devansu-Yadav/eslint-plugin-import/blob/main/docs/rules/newline-after-import.md#rule-details) to group together comments along with imports without any newline or it would deviate the purpose of this rule. I'm open to any suggestions for the schema we should have for this rule.

Any thoughts on this? @ljharb

**Comment by ghmcadams** [comment]
I thought I would weigh in on this one - I would love if this were part of the `order` rule.

I see a couple of potential options:

1. The `newlines-between` option could be modified to allow a numeric value
1. Another property could be added (`spacing-between: number`?), which would only be used when `newlines-between` has a value of `always' or 'always-and-inside-groups`
